python-checks/src/test/resources/checks/argumentType.py

```diff
@@ -59,25 +59,27 @@ def stdlib_functions():
   A = UnexpectedClass()
   acos(A) # Noncompliant
   B = datetime.tzinfo()
-  B.tzname(42) # Noncompliant {{Change this argument; Function "tzname" expects a different type}}
-#          ^^
+  B.tzname(42) # FN Noncompliant {{Change this argument; Function "tzname" expects a different type}}
+##          ^^
   select.select([],[],[], 0)
   time.sleep(1) # OK
   x = time.gmtime(int(time.time()))
   x = time.gmtime(secs = int(time.time()))
   time.sleep(True) # OK, converted to 1
   time.sleep(1j) # FN, considered duck type compatible
   genericpath.isfile("some/path")
-  genericpath.isfile(42) # Noncompliant
+  genericpath.isfile(42) # FN Noncompliant
```

Comment:
The expected parameter is now an union between the following types: `bytes`, ``int`, `str` and `os.PathLike`.
Interestingly, I don't get a `TypeError` when executing this, it simply returns `False` (Python 3.9).
We should look if this API has changed. From a first look, this looks like a legitimate change.

Reply:
Not much has changed in this method. It uses os.stat which when provided with an int throws an OSError. The error is then caught by genericpath.isfile and returns None. So I would agree here the program is safe but it should always return False if we pass an int to genericpath.isfile.